src/main/java/uk/gov/hmcts/darts/task/runner/impl/AssociatedObjectDataExpiryDeletionAutomatedTask.java

```diff
@@ -168,6 +177,32 @@ <T extends SoftDelete & HasIntegerId> void deleteExternalObjectDirectoryEntity(
         entitiesToDelete.forEach(t -> auditApi.record(auditActivity, userAccount, String.valueOf(t.getId())));
     }
 
+    <T extends SoftDelete & HasIntegerId & HasRetention & CanReturnExternalObjectDirectoryEntities> boolean shouldDeleteFilter(T entity) {
+        ExternalObjectDirectoryEntity armExternalObjectDirectoryEntity = entity.getExternalObjectDirectoryEntities()
+            .stream()
+            .filter(e -> ExternalLocationTypeEnum.ARM.getId().equals(e.getExternalLocationType().getId()))
+            .findFirst()
+            .orElse(null);
+
+        if (armExternalObjectDirectoryEntity == null) {
+            log.debug("Skipping deletion of {} with id {} as there is no ARM external object directory entity",
```

Comment:
Should these be info statements? If these are skipped then you probably want to know about it.